### ISSUE TITLE
torch.sigmoid and torch.nn.sigmoid are different (other functions too?)

### ISSUE THREAD

**Issue by csarofeen** [pytorch/pytorch#1245]
Performance and numerics of torch.sigmoid and torch.nn.functional.sigmoid are different kernels. One being...
```.cpp
template <typename T>
struct TensorSigmoidOp {
  __device__ __forceinline__ void operator()(T* out, T* in) const {
    T one = (T) 1.0;
    *out = one / (one + THCNumerics<T>::exp(- *in));
  }

  __device__ __forceinline__ void operator()(T* v) const {
    T one = (T) 1.0;
    *v = one / (one + THCNumerics<T>::exp(- *v));
  }
};
```

The other being...
```.cpp
template <typename T>
struct sigmoidupdateOutput_functor
{
  __device__ void operator()(T *output, const T *input) const
  {
    *output = ScalarConvert<double, T>::to(1./(1.+ exp(-*input)));
  }
};
```

Typical numerical difference between sigmoid functions appears to be ~5.96e-8 . Performance difference could be significant due to the double precision operations.

Should they be different?
Which one is preferred?
How wide spread is this?
Shouldn't THCTensorMathPointwise functions be used in THCUNN operations when possible?

Opinions @gchanan @jekbradbury ?

**Comment by apaszke** [comment]
I think we should move the forward implementation to THC and use the backward code form THCUNN in autograd. Using double precision seems unnecessary to me.
